tests/test_api.py

```diff
@@ -983,7 +983,7 @@ def test_api_cancel_build(testclient):
         if r.data.status == schema.BuildStatus.BUILDING.value:
             building = True
             break
-        time.sleep(30)
+        time.sleep(5)
```

Comment:
I've seen the build status being set to COMPLETED because this was first checking too early and then too late.